### PR TITLE
Update TFM in tests to net8.0

### PR DESCRIPTION
**Pull request by jjonescz**
https://github.com/dotnet/roslyn/pull/69506#discussion_r1300755318

### CODE REVIEW

File: src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer.UnitTests/Microsoft.CodeAnalysis.LanguageServer.UnitTests.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net7.0</TargetFramework>
+    <TargetFramework>net8.0</TargetFramework>
```
Comment (genlu) [inline src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer.UnitTests/Microsoft.CodeAnalysis.LanguageServer.UnitTests.csproj:4]: This should remain 7.0.

Comment (jjonescz) [reply src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer.UnitTests/Microsoft.CodeAnalysis.LanguageServer.UnitTests.csproj:4]:
```suggestion
    <TargetFramework>net7.0</TargetFramework>
```

**Review by 333fred — APPROVED** [review]

File: src/Compilers/VisualBasic/Test/Emit/Attributes/AttributeTests_WellKnownAttributes.vb [hunk 2]
```diff
@@ -15,6 +15,8 @@ Imports Microsoft.CodeAnalysis.VisualBasic.Symbols.Metadata.PE
 Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
 Imports Roslyn.Test.Utilities
 
+#Disable Warning SYSLIB0050 ' 'TypeAttributes.Serializable' is obsolete
```
Comment (jcouv) [inline src/Compilers/VisualBasic/Test/Emit/Attributes/AttributeTests_WellKnownAttributes.vb:18]: What's the context for this? I'm not seeing that attribute as obsolete on [https://learn.microsoft.com/en-us/dotnet/api/system.reflection.typeattributes?view=net-8.0](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.typeattributes?view=net-8.0)
If it is obsolete, are we supposed to use something else instead?

**Review by jcouv — COMMENTED** [review]
Done with review pass (iteration 2). I just have a question on the obsoleted attribute

File: src/Compilers/VisualBasic/Test/Emit/Attributes/AttributeTests_WellKnownAttributes.vb (hunk 2 above)
Comment (jjonescz) [reply src/Compilers/VisualBasic/Test/Emit/Attributes/AttributeTests_WellKnownAttributes.vb:18]: It's suppressed at other places that were upgraded to net8 previously, e.g.:

https://github.com/dotnet/roslyn/blob/32b3fc05bff6daca73d91c3ab162c2aa7a3cb7aa/src/Compilers/VisualBasic/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.vb#L223

See also comment in https://github.com/dotnet/roslyn/pull/68627/files

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 2)

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 2)

**Pull request merged**